jscomp/core/js_exp_make.ml

```diff
@@ -810,7 +812,8 @@ let uint32 ?comment n : J.expression =
 
 let string_comp (cmp : J.binop) ?comment (e0 : t) (e1 : t) =
   match (e0.expression_desc, e1.expression_desc) with
-  | Str {txt=a0}, Str {txt=b0} -> (
+  | Str {txt=a0}, Str {txt=b0}
+  | Unicode a0, Unicode b0 -> (
```

Comment:
Equality s1 == s2 should correspond to equal byte representation also in unicode.